### PR TITLE
Add dag-only deploy feature (part 1)

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

- Add dag-deploy chart features and tests
- Improve helm_template_generator.py by using modern python syntax, `DEBUG=1` now keeps the values files when running tests.
- Update docs

## Related Issues

- https://github.com/astronomer/issues/issues/5970

## Testing

Testing steps included in ticket about the working of the feature. Unit tests are included in this PR, but QA will need to do extensive testing on this feature and we will surely need to iterate on it.

## Merging

This is only for release-0.34

### CODE REVIEW

File: bin/reset-local-dev [hunk 1]
```diff
@@ -15,5 +15,7 @@ helm install \
   -n airflow \
   --set executor="$EXECUTOR" \
   --set workers.persistence.fixPermissions=true \
+  --set dagServer.enabled=True \
```
Comment (danielhoherd) [inline bin/reset-local-dev:18]: We need to delete line 18 before merging. Line 19 is fine to keep.

File: dag-server-test-values.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# !!!!!!!!!!!!!!!!!
+#
+# TEMP FILE FOR TESTING. DELETE BEFORE MERGING TO MASTER
+#
+# !!!!!!!!!!!!!!!!!
+
+airflow:
+  executor: CeleryExecutor
```
Comment (danielhoherd) [inline dag-server-test-values.yaml:8]: This file should be deleted before merging.

**Comment by rishkarajgi** [comment]
Is this WIP?

**Comment by danielhoherd** [comment]
@pgvishnuram do you have any more changes for the airfow-chart or are we good to mark this PR draft as ready for review?

File: dag-server-test-values.yaml (hunk 2 above)
Comment (pgvishnuram) [reply dag-server-test-values.yaml:8]: Im deleting this file to merge this change to master

**Review by pgvishnuram — APPROVED** [review]

**Comment by pgvishnuram** [comment]
we can even add a standalone documentation how customer can integrate this feature standalone

File: bin/reset-local-dev (hunk 1 above)
Comment (pgvishnuram) [reply bin/reset-local-dev]: removed dagServer.enabled

**Pull request merged**